src/util/triplea/MapXMLCreator/CanalDefinitionsPanel.java

```diff
@@ -0,0 +1,442 @@
+        }
+        if (prevCanalCount > 0 || !noNewCanalsBuild)
+          panel.repaint();
+      }
+    });
+  }
+
+  protected void paintCenterSpecifics(final Graphics g, final String centerName, final FontMetrics fontMetrics,
+      final Point item, final int x_text_start) {
+    if (selectedLandTerritories.contains(centerName) || selectedWaterTerritories.contains(centerName)) {
+      final Rectangle2D stringBounds = fontMetrics.getStringBounds(centerName, g);
+      g.setColor(Color.yellow);
+      g.fillRect(Math.max(0, x_text_start - 2), Math.max(0, item.y - 6), (int) stringBounds.getWidth() + 4,
+          (int) stringBounds.getHeight());
+      g.setColor(Color.red);
+      g.drawString(centerName, Math.max(0, x_text_start), item.y + 5);
+    }
+    g.setColor(Color.red);
+  }
+
+  protected void paintPreparation(final Map<String, Point> centers) {}
+
+  @Override
+  protected void paintOwnSpecifics(Graphics g, Map<String, Point> centers) {
+    Graphics2D g2d = (Graphics2D) g;
+    for (final Entry<String, Tuple<SortedSet<String>, SortedSet<String>>> canalDef : MapXMLHelper.canalDefinitions
+        .entrySet()) {
+      // Water Territories
+      g.setColor(Color.BLUE);
+      final String canalName = canalDef.getKey();
+      final FontMetrics fontMetrics = g.getFontMetrics();
+      SortedSet<String> terrSet1 = canalDef.getValue().getFirst();
+      SortedSet<String> remainingTerrs = new TreeSet<String>(terrSet1);
+      for (final String terr1 : terrSet1) {
+        final Point center1 = centers.get(terr1);
+        remainingTerrs.remove(terr1);
+        for (final String terr2 : remainingTerrs) {
+          final Point center2 = centers.get(terr2);
+          g.drawLine(center1.x, center1.y, center2.x, center2.y);
+          Rectangle2D stringBounds = fontMetrics.getStringBounds(canalName, g);
+          int dX = center2.x - center1.x;
+          int dY = center2.y - center1.y;
+          final Point lineCenter = new Point(center1.x + dX / 2, center1.y + dY / 2);
+          double centerDistance = center2.distance(center1);
+          if (centerDistance > stringBounds.getWidth())
+            drawRotate(g2d, lineCenter.x, lineCenter.y, Math.atan2(dY, dX), canalName,
+                (int) (stringBounds.getWidth()) / -2);
+          else
+            g.drawString(canalName, lineCenter.x, lineCenter.y);
+        }
+      }
+      // Land Territories
+      g.setColor(Color.GREEN);
+      terrSet1 = canalDef.getValue().getSecond();
+      remainingTerrs = new TreeSet<String>(terrSet1);
+      for (final String terr1 : terrSet1) {
+        final Point center1 = centers.get(terr1);
+        remainingTerrs.remove(terr1);
+        for (final String terr2 : remainingTerrs) {
+          final Point center2 = centers.get(terr2);
+          g.drawLine(center1.x, center1.y, center2.x, center2.y);
+        }
+      }
+    }
+
+  }
+
+  static final double piHalf = Math.PI / 2;

```

Comment:
This is out of place, top of file for constants, and should be private unless package protected is the minimum scope possible for it.